### PR TITLE
[MM-50534] - Renew email adds portal link when not eligible for self-serve renewal

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
When license is about to expire or has expired we send reminder emails for admins to renew. This ticket makes a check if an admin can renew a license from the portal, if they can we add a CTA to renew from portal, otherwise the CTA is to contact sales

As part of this PR the following change has been made:
We need to make 2 checks about the self-serve license, is it renewable and are the seats expandable. Instead of having different endpoints to make these 2 checks, on the self-serve portal, there's now one endpoint that can help give information about whether the license is renewable and expandable. Any new information about the self-serve status of a subscription or license can easily be added to this endpoint without need to create a new one.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50534

#### Related Pull Requests

https://github.com/mattermost/enterprise/pull/1384
https://github.com/mattermost/customer-web-server/pull/1116

#### Screenshots
Cannot renew from portal
---
Screenshot 2023-02-21 at 16 35 33

Screenshot 2023-02-21 at 16 59 15

Can renew from portal
---
Screenshot 2023-02-21 at 16 38 45

Screenshot 2023-02-21 at 17 02 22
Screenshot 2023-02-21 at 17 02 22

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by ConorMacpherson — APPROVED** [review]
LGTM

**Comment by yasserfaraazkhan** [comment]
@AGMETEOR ,
to test this, we need to point `renewalURL` to `portal.test.cloud` , right ?

File: app/server.go [hunk 1]
```diff
@@ -1380,9 +1400,16 @@ func (s *Server) doLicenseExpirationCheck() {
 			continue
 		}
 
+		T := i18n.GetUserTranslations(user.Locale)
+		ctaText := T("api.templates.license_up_for_renewal_renew_now")
+		if !status.IsExpandable {
```
Comment (nickmisasi) [inline app/server.go:1405]: IsExpandable looks to rely on [the subscription status being active](https://github.com/mattermost/customer-web-server/blob/master/model/subscription.go#L255) which will never actually be the case after the license has expired (we cancel them at period end), so i don't think this path will ever fire

**Review by nickmisasi — APPROVED** [review]
Thanks Allan!

**Review by neallred — APPROVED** [review]
Ditto LGTM pending whatever we decide on "checks" vs "status" naming and on using hosted_customer vs cloud files.

**Review by ConorMacpherson — APPROVED** [review]

**Review by wallyf1972 — APPROVED** [review]

**Comment by yasserfaraazkhan** [comment]
@AGMETEOR , was not able to test this on the PR https://github.com/mattermost/mattermost-server/pull/22367 as spin wick  is running team edition.
I can test it post merge.

**Comment by amyblais** [comment]
> Hey @amyblais the cherry-pick should happen to cloud branch, right?

@AGMETEOR Maria had added the 7.9 milestone to the Jira ticket https://mattermost.atlassian.net/browse/MM-50534 so I thought this goes to self-hosted. Is this for cloud instead?

**Comment by amyblais** [comment]
@AGMETEOR ``release-7.9`` branch for server/webapp/enteprise https://github.com/mattermost/mattermost-webapp/pull/12263#issuecomment-1453709417. Please open PRs.

**Pull request merged**